libmscore/chordrest.cpp

```diff
@@ -1232,20 +1231,31 @@ QString ChordRest::accessibleExtraInfo() const
 
 bool ChordRest::isMelismaEnd() const
       {
-      return _melismaEnd;
+      return !_melismaEnds.empty();
       }
 
 //---------------------------------------------------------
-//   setMelismaEnd
+//   removeMelismaEnd()
+//    removes a Lyrics object from the _melismaEnds set
+//    if present by iterating through the set.
+//    Returns a boolean representing whether the Lyrics
+//    was found.
+//    No-op and return false if Lyrics not present in _melismaEnds.
 //---------------------------------------------------------
 
-void ChordRest::setMelismaEnd(bool v)
+bool ChordRest::removeMelismaEnd(const Lyrics* l)
```

Comment:
just use the combination of "std::set<Key,Compare,Allocator>::find" and "std::set<Key,Compare,Allocator>::erase"

I'm also not sure if whether it makes sense to store it in std::set, what was the point?

Reply:
Good call; I honestly can't recall why I made this so overly complex.
The reason I'm storing the values is this: previously there was a _melismaEnd boolean that stored whether a ChordRest had the end of a melisma, but this actually was unreliable if there were multiple lyrics. Additionally, this spacing fix required that a ChordRest have pointers to any lyrics/melismas that end on it in order to create a properly-sized spacer. Since I needed to store those pointers and there already was an existing melisma-end data member, I basically made a drop-in replacement for the existing _melismaEnd functions that stored all pointers (realistically, this will be ~0–3 values) rather than just setting a boolean. I chose a std::set to avoid redundancy (since I didn't write the original code where they get added); although possibly an unordered_set would be quicker, or some other Qt container. Does that make sense? Let me know if I'm missing something

Reply:
ah, sorry, missed that one in my emails

prefer to use std::containers wherever it's possible within the libmscore
regarding your case, I think it's enough to stick with std::set then